docker/android-system.sh

```diff
@@ -46,64 +79,474 @@ EOF
+            sync prebuilts/gcc/linux-x86/aarch64/aarch64-linux-android-4.9
+        ;;
+        x86)
+            sync prebuilts/gcc/linux-x86/x86/x86_64-linux-android-4.9
+        ;;
+        x86_64)
+            sync prebuilts/gcc/linux-x86/x86/x86_64-linux-android-4.9
+        ;;
+    esac
+
+    # avoid build tests
+    rm bionic/linker/tests/Android.mk
+    rm bionic/tests/Android.mk
+    rm bionic/benchmarks/Android.mk
+    rm prebuilts/misc/common/android-support-test/Android.mk
+    # we don't need the relocation packer, and removing
+    # the unittests from it is a bit of work.
+    rm bionic/tools/relocation_packer/Android.mk
+
+    remove_tests
+}
+
+# tested on 8.0.0_r51 (SDK 26)
+# tested on 8.1.0_r81 (SDK 27)
+android_repo_v8() {
+    # need to build LLVM components, or libLLVM is disabled.
+    export FORCE_BUILD_LLVM_COMPONENTS=true
+
+    sync bionic
+    sync build/blueprint
+    sync build/make
+    sync build/soong
+    sync external/boringssl
+    sync external/clang
+    sync external/compiler-rt
+    sync external/elfutils
+    sync external/jemalloc
+    sync external/libcxx
+    sync external/libcxxabi
+    sync external/libevent
+    sync external/libunwind
+    sync external/libunwind_llvm
+    sync external/llvm
+    sync external/lzma
+    sync external/mksh
+    sync external/pcre
+    sync external/safe-iop
+    sync external/selinux
+    sync external/zlib
+    sync libnativehelper
+    sync prebuilts/build-tools
+    sync prebuilts/clang/host/linux-x86
+    sync prebuilts/gcc/linux-x86/host/x86_64-linux-glibc2.15-4.8
+    sync prebuilts/go/linux-x86
+    # we only need the relocation packer binary. everything else
+    # interferes with the build, so we remove the makefiles below.
+    sync prebuilts/misc
+    sync prebuilts/ndk
+    sync system/core
+    sync toolchain/binutils
+
+    case "${ARCH}" in
+        arm)
+            sync prebuilts/gcc/linux-x86/arm/arm-linux-androideabi-4.9
+        ;;
+        arm64)
+            sync prebuilts/gcc/linux-x86/arm/arm-linux-androideabi-4.9
+            sync prebuilts/gcc/linux-x86/aarch64/aarch64-linux-android-4.9
+        ;;
+        x86)
+            sync prebuilts/gcc/linux-x86/x86/x86_64-linux-android-4.9
+        ;;
+        x86_64)
+            sync prebuilts/gcc/linux-x86/x86/x86_64-linux-android-4.9
+        ;;
+    esac
+
+    # avoid build tests
+    rm bionic/linker/tests/Android.mk
+    rm bionic/tests/Android.mk
+    rm bionic/tests/Android.bp
+    rm bionic/benchmarks/Android.bp
+    rm bionic/tests/libs/Android.bp
+
+    # remove extra utilities
+    rm system/core/libgrallocusage/Android.bp
+    rm system/core/libmemtrack/Android.bp
+    rm system/core/libsysutils/Android.bp
+    local path=
+    find prebuilts/misc/ -name 'Android.mk' | while IFS= read -r path; do
+        rm "${path}"
+    done
+
+    # avoid java dependencies
+    rm external/lzma/Java/Tukaani/Android.mk
+
+    remove_tests
```

Comment:
Due to the introduction of the Soong build system in Android 8+, we need much more complex build system modification. This is described in detail in `android/soong.py` below.